pkg/autoupgrade/daemon_test.go

```diff
@@ -337,6 +341,30 @@ func TestRefreshImages(t *testing.T) {
 			appKeysPrevCheckAfter:  map[kclient.ObjectKey]time.Time{router.Key("acorn", "notify-app"): now},
 			appsUpdated:            map[string]string{"notify-app": "docker.io/acorn/notify"},
 		},
+		{
+			name:                   "Auto refresh tag with multiple remote tags with latest tag image denied and current being an older image",
+			client:                 &mockDaemonClient{remoteTags: []string{"v1.1.1", "v1.1.2"}, imageDenyList: map[string]struct{}{"index.docker.io/acorn/test-1:v1.1.2": {}}},
+			appKeysPrevCheckBefore: map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): thirtySecondsAgo},
+			imagesToRefresh:        map[imageAndNamespaceKey][]kclient.ObjectKey{{image: "index.docker.io/acorn/test-1:v1.1.1", namespace: "acorn"}: {router.Key("acorn", "test-1")}},
+			appKeysPrevCheckAfter:  map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): now},
+			appsUpdated:            map[string]string{"test-1": "index.docker.io/acorn/test-1:v1.1.1"},
+		},
+		{
+			name:                   "Auto refresh tag with multiple remote tags with latest tag image denied and current being latest image",
+			client:                 &mockDaemonClient{remoteTags: []string{"v1.1.1", "v1.1.2"}, imageDenyList: map[string]struct{}{"index.docker.io/acorn/test-1:v1.1.2": {}}},
+			appKeysPrevCheckBefore: map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): thirtySecondsAgo},
+			imagesToRefresh:        map[imageAndNamespaceKey][]kclient.ObjectKey{{image: "index.docker.io/acorn/test-1:v1.1.2", namespace: "acorn"}: {router.Key("acorn", "test-1")}},
+			appKeysPrevCheckAfter:  map[kclient.ObjectKey]time.Time{router.Key("acorn", "test-1"): now},
+			appsUpdated:            map[string]string{"test-1": "index.docker.io/acorn/test-1:v1.1.2"},
```

Comment:
This is saying that you expect the app`test-1` to be updated to image `index.docker.io/acorn/test-1:v1.1.2` in this scenario.

But since `v1.1.2` is on the imageDenyList, that is not the desired outcome in this scenario. The desired outcome is that  the app is not updated (and so, the `appsUpdated` map should be empty).

Reply:
that makes sense, I was under the assumption that appsUpdated will need some value to check against with. But essentially there should be no change.